### PR TITLE
Add RXT templates per version

### PR DESCRIPTION
**Pull request by HeshanSudarshana**
## Purpose
This PR adds the following changes,
- [x] Remove UMA tables from 2.6.0 to 3.0.0 pre-migration scripts
- [x] Add API RXT templates per version
- [x] Remove unnecessary API_UUID query in pre-migration scripts from earlier versions than 4.0.0
- [x] Remove setting enable store property from V300 migrator since the attribute is introduced in 3.2.0

### CODE REVIEW

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+public class ArtifactReIndexingMigrator extends Migrator {
+    private static final Log log = LogFactory.getLog(ArtifactReIndexingMigrator.class);
+    private final String artifactReIndexingScriptPath = Utility.POST_MIGRATION_SCRIPT_DIR + "common" + File.separator
+            + "reg_db" + File.separator + "reg-index.sql";
+
+    @Override
+    public void migrate() throws APIMigrationException {
+        try {
+            AMDBUtil.runSQLScript(artifactReIndexingScriptPath, true);
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java:21]: @HeshanSudarshana  Shall we add more info logs for this phase (Start ArtifactReIndexingMigrator and Finish ArtifactReIndexingMigrator) ?

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import org.wso2.carbon.apimgt.migration.migrator.Migrator;
+import org.wso2.carbon.apimgt.migration.migrator.Utility;
+import org.wso2.carbon.apimgt.migration.util.AMDBUtil;
+
+import java.io.File;
+import java.sql.SQLException;
+
+public class ArtifactReIndexingMigrator extends Migrator {
+    private static final Log log = LogFactory.getLog(ArtifactReIndexingMigrator.class);
+    private final String artifactReIndexingScriptPath = Utility.POST_MIGRATION_SCRIPT_DIR + "common" + File.separator
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java:15]: shall we move all the constants to the constant file?

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/client/V410Migration.java [hunk 3]
```diff
@@ -17,9 +17,9 @@ public class V410Migration extends VersionMigrator {
     private static final Log log = LogFactory.getLog(V410Migration.class);
     private final String PRE_MIGRATION_SCRIPTS_PATH = Utility.PRE_MIGRATION_SCRIPT_DIR + "migration-4.0.0_to_4.1.0"
             + File.separator;
-    private final String POST_MIGRATION_SCRIPT_REGDB_PATH = Utility.POST_MIGRATION_SCRIPT_DIR + "reg_db" + File.separator
-            + "reg-index.sql";
-    private final String POST_MIGRATION_SCRIPT_AMDB_PATH = Utility.POST_MIGRATION_SCRIPT_DIR + "am_db" + File.separator;
+    private final String POST_MIGRATION_SCRIPT_AMDB_PATH = Utility.POST_MIGRATION_SCRIPT_DIR
+            + "migration-4.0.0_to_4.1.0" + File.separator + "am_db" + File.separator;
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/client/V410Migration.java:21]: shall we move this to the same constant file

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.migration.APIMigrationException;
+import org.wso2.carbon.apimgt.migration.migrator.Migrator;
+import org.wso2.carbon.apimgt.migration.migrator.Utility;
+import org.wso2.carbon.apimgt.migration.util.AMDBUtil;
+
+import java.io.File;
+import java.sql.SQLException;
+
+public class ArtifactReIndexingMigrator extends Migrator {
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java:13]: Shall we add class level comment on what this migrator does?

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/RegistryResourceMigrator.java [hunk 5]
```diff
@@ -20,10 +21,12 @@ public class RegistryResourceMigrator extends Migrator {
     private static final Log log = LogFactory.getLog(RegistryResourceMigrator.class);
     private RegistryService registryService;
     List<Tenant> tenants;
+    private String rxtDir;
 
-    public RegistryResourceMigrator() throws UserStoreException {
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/RegistryResourceMigrator.java:24]: Shall we add a class level comment on the purpose of this migrator?

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/RegistryResourceMigrator.java [hunk 6]
```diff
@@ -34,26 +37,21 @@ public void migrate() throws APIMigrationException {
...
 
         for (Tenant tenant : tenants) {
             try {
                 regService.startTenantFlow(tenant);
 
                 log.info("Updating api.rxt for tenant " + tenant.getId() + '(' + tenant.getDomain() + ')');
                 //Update api.rxt file
-                String rxt = FileUtil.readFileToString(rxtDir);
-                regService.updateRXTResource(rxtName, rxt);
+                String rxt = FileUtil.readFileToString(rxtPath);
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/RegistryResourceMigrator.java:48]: better to log the rxtPath as well.

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
+package org.wso2.carbon.apimgt.migration.migrator.commonMigrators;
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java:1]: license header missing

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java (hunk 1 above)
Comment (HeshanSudarshana) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java]: Added with https://github.com/wso2-extensions/apim-migration-resources/pull/182/commits/72b02fe75207489c6883097a9ebd042594432a12

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java (hunk 2 above)
Comment (HeshanSudarshana) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java]: Added with https://github.com/wso2-extensions/apim-migration-resources/pull/182/commits/33b22bdc4d5975483399ed88ca62216fd0d504d4

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/client/V410Migration.java (hunk 3 above)
Comment (HeshanSudarshana) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/client/V410Migration.java]: Added with https://github.com/wso2-extensions/apim-migration-resources/pull/182/commits/33b22bdc4d5975483399ed88ca62216fd0d504d4

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java (hunk 4 above)
Comment (HeshanSudarshana) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java:31]: Added with https://github.com/wso2-extensions/apim-migration-resources/pull/182/commits/46c296a100d1609a5bff121246b7801e44c62ad4

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/RegistryResourceMigrator.java (hunk 5 above)
Comment (HeshanSudarshana) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/RegistryResourceMigrator.java:24]: Added with https://github.com/wso2-extensions/apim-migration-resources/pull/182/commits/46c296a100d1609a5bff121246b7801e44c62ad4

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java (hunk 7 above)
Comment (HeshanSudarshana) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/ArtifactReIndexingMigrator.java:17]: Added with https://github.com/wso2-extensions/apim-migration-resources/pull/182/commits/11df4515226a79232461d6feae7669b9c2339527

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/RegistryResourceMigrator.java (hunk 6 above)
Comment (HeshanSudarshana) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/commonMigrators/RegistryResourceMigrator.java:66]: Added with https://github.com/wso2-extensions/apim-migration-resources/pull/182/commits/d49880b0ef5ca86b814957a6ed36db6e0465dde4

**Pull request merged**